configure.ac

```diff
@@ -62,28 +62,33 @@ AS_IF([test "x$enable_openmp" != xno], [
     AC_ARG_ENABLE([nestedomp], [AS_HELP_STRING([--enable-nestedomp],
                   [build with nested OMP capability])])
     test "x$enable_nestedomp" = xyes || enable_nestedomp=no
+    if test "x$enable_nestedomp" = xyes; then
+     AC_DEFINE([ENABLE_NESTEDOMP], [1], [nested omp capability enabled])
+    fi
     AC_MSG_RESULT($enable_nestedomp)
   fi
-])
+fi
 AM_CONDITIONAL(ENABLE_NESTEDOMP, [test x$enable_nestedomp = xyes])
 
-# Check for pthread library only if --disable-openmp was specified
-AS_IF([test "$useomp" = no], [
-dnl Example of default-disabled feature
-  AC_ARG_ENABLE([pthread],
-      AS_HELP_STRING([--enable-pthread], [Enable pthread capability]))
-
-  AS_IF([test "x$enable_pthread" = "xyes"], [
-    dnl Do the stuff needed for enabling the feature
-    AC_MSG_CHECKING([for pthread support])
-    AC_CHECK_LIB([pthread], [pthread_mutex_init])
-    if test "x$ac_cv_lib_pthread_pthread_mutex_init" = xyes; then
-       AC_DEFINE([THREADED_PTHREADS], [1], [pthreads library is present])
-    fi
-    AC_MSG_RESULT([$enable_pthread])
-  ])
-  
-])
+# pthread library. default disabled
+AC_ARG_ENABLE([pthread],
+    AS_HELP_STRING([--enable-pthread], [Enable pthread capability]))
+
+if test "x$enable_pthread" = xyes; then
```

Comment:
At this point $enable_pthread will be unset if the user accepts the default. This leads to confusing errors. What I have done is settled on this idiom, which I would like to use in the GPTL configure.ac:

```
# Does the user want to enable logging?
AC_MSG_CHECKING([whether debug logging is enabled])
AC_ARG_ENABLE([logging],
              [AS_HELP_STRING([--enable-logging],
                              [enable debug logging capability (will negatively impact performance). \
			      This debugging feature is probably only of interest to PIO developers.])])
test "x$enable_logging" = xyes || enable_logging=no
AC_MSG_RESULT([$enable_logging])
```

Note that after this code is run, $enable_logging is always defined, to either "yes" or "no". This is useful, for example in the AC_MSG_RESULT call, which provides valuable feedback in the configure output, to help with support.

The way this has to be done changes a little when the features is enabled by default, instead of disabled by defualt. Here's one of those:

```
# Does the user want to disable pnetcdf?
AC_MSG_CHECKING([whether pnetcdf is to be used])
AC_ARG_ENABLE([pnetcdf],
              [AS_HELP_STRING([--disable-pnetcdf],
                              [Disable pnetcdf use.])])
test "x$enable_pnetcdf" = xno || enable_pnetcdf=yes
AC_MSG_RESULT([$enable_pnetcdf])

```

(Note that the AC_ARG_ENABLE call indicates which is the default by providing the non-default option.)